### PR TITLE
Refactor i18n #519

### PR DESCRIPTION
**Pull request by kani3camp**
This pull request introduces a new type-safe, auto-generated i18n translation system, refactors the i18n package to use a dedicated internal engine, and updates tests and documentation to reflect these changes. The new system generates Go wrapper functions for each translation key based on TOML locale and meta files, ensuring compile-time safety for translation arguments. Legacy translation APIs are removed, and the codebase is reorganized for clarity and maintainability.

### i18n System Refactor & Auto-Generation

**Introduction of auto-generated, type-safe i18n translation functions:**
* Added a generator (`system/cmd/i18n-gen/main.go`) that reads TOML locale and meta files and produces type-safe wrapper functions for each translation key in `core/i18n/typed/zz_generated.i18n_messages.go`. This ensures argument count/type safety at compile time and replaces the old `i18n.T(...)` API. [[1]](diffhunk://#diff-61311d19d16959656901907fd7ffcc8c290087419344d1ee0d1c63eebeff3bf4R1-R322) [[2]](diffhunk://#diff-69e9b086a3f0f8cf989e3fe9483e10be2d558ffeeca1d1f0f7d95da1ac4db994L2-R39) [[3]](diffhunk://#diff-c9694af2e6ee937160488d55b4cd408113fd009fa9cf76a65a4e673f97819da8R1-R3)

**Refactoring of the public i18n API to use an internal engine:**
* The main `i18n` package now delegates all translation logic and data management to `internal/engine`, exposing only minimal API and types. Legacy translation functions, localizer types, and argument formatting logic have been removed. [[1]](diffhunk://#diff-736e5c600b2700d28308159077138add5ebe900efb4811b63740be7bdfcc3b6bL5-R36) [[2]](diffhunk://#diff-cfd15232f597050bf23aeec86de957cbdba0b2c64585f061bd01852f95cbd75cR1-R120)

### Testing & Documentation

**Comprehensive update and expansion of i18n tests:**
* The test suite (`system/core/i18n/i18n_test.go`) has been rewritten to verify real-world usage, parameter replacement, fallback logic, concurrent access, and edge cases using both the engine and the new generated functions.

**Documentation improvements:**
* The README now documents the new i18n auto-generation workflow and usage, replacing outdated instructions.

### Cleanup

**Removal of obsolete test locale file:**
* Deleted the old English test locale file (`system/core/i18n/en.toml`) that is no longer needed.

### Miscellaneous

**VSCode settings update:**
* Added `"usecase"` to the list of recognized code folders in `.vscode/settings.json`.

### CODE REVIEW

**Comment by kani3camp** [comment]
レビュー指摘に対応しました。\n- presenter: `in.go`, `more.go`, `change.go` で文字列連結を `strings.Builder` に変更\n- i18n-gen: warn文言の重複(warn:)を削除し簡潔化\n- usecase: `events.go` にパッケージレベルの説明コメントを追加（イベント駆動とPresenter分離の意図を明記）\n\n全テスト `go test ./...` はグリーンです。\nご確認お願いします。

**Comment by kani3camp** [comment]
追加コメントに対応・確認しました。\n\n- Inの`result` 明示初期化: 可読性のため修正済み\n- sir接頭辞の仕様: 実装・テストの意図は以下で統一です\n  - Break/Order/Resume/Clear: 基本はsirなし。EnterOnly時のみ `sir + EnterOnly`\n  - Change/More: sirあり\n  - コード上の例: BreakEnterOnlyは `CommonSir + CommandEnterOnly`、その他のBreakケースはsirなし\n- i18n-gen: `I18N_STRICT=1` で欠落/不一致をpanicに昇格（デフォルトはwarnのまま運用可能）。CIで厳格化する場合は環境変数を有効にしてください。\n- TODO指摘について: `command_seat.go` 内に該当のTODOは残っておらず、`ValidateChange` は実装済みです（該当箇所は通常のエラーハンドリングのみ）。\n\n`go test ./...` は引き続きグリーンです。追加の修正要望があればご指摘ください。

**Comment by kani3camp** [comment]
読みやすい形式で再掲します。

- In の result 明示初期化: 修正済み
- sir 接頭辞の仕様（実装・テストの意図）
  - Break/Order/Resume/Clear: 基本は sir なし。EnterOnly のときのみ「sir + EnterOnly」
  - Change/More: sir あり
  - コード例: BreakEnterOnly は `CommonSir + CommandEnterOnly`、その他の Break ケースは sir なし
- i18n-gen
  - `I18N_STRICT=1` で欠落/不一致を panic に昇格（デフォルトは warn のまま運用可能）
  - CI で厳格化する場合は環境変数を有効化してください
- TODO 指摘について
  - `command_seat.go` に該当の TODO はなく、`ValidateChange` は実装済み（通常のエラーハンドリングのみ）
- テスト
  - `go test ./...` は引き続きグリーンです

上の改行が崩れたコメントは読みづらくしてしまい失礼しました。以後は本文ファイル経由で投稿します。

File: system/core/workspaceapp/command_seat.go [hunk 1]
```diff
@@ -272,30 +277,31 @@ func (app *WorkspaceApp) In(ctx context.Context, inOption *utils.InOption) error
...
-
-			// 入室しましたのメッセージ
-			replyMessage += t("start", app.ProcessedUserDisplayName, inOption.MinWorkOrderOption.WorkName, untilExitMin, newSeatId)
+			result.Add(usecase.SeatEntered{
+				SeatID:       inOption.SeatId,
+				IsMemberSeat: isTargetMemberSeat,
+				WorkName:     inOption.MinWorkOrderOption.WorkName,
+				UntilExitMin: untilExitMin,
+			})
+		}
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/command_seat.go:286]: 問題なし

File: system/core/workspaceapp/presenter/break.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		switch e := event.(type) {
+		case usecase.BreakEnterOnly:
+			msg += i18nmsg.CommonSir(displayName) + i18nmsg.CommandEnterOnly()
+		case usecase.BreakWorkOnly:
+			msg += i18nmsg.CommandBreakWorkOnly(displayName)
+		case usecase.BreakWarn:
+			msg += i18nmsg.CommandBreakWarn(displayName, e.MinBreakIntervalMin, e.CurrentWorkedMin)
+		case usecase.BreakStarted:
+			seat := SeatIDStr(e.SeatID, e.IsMemberSeat)
+			msg += i18nmsg.CommandBreakBreak(displayName, e.WorkName, e.DurationMin, seat)
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/presenter/break.go:22]: これは現実装のe.DurationMinであっています。ユーザーが指定した時間が最長になります。

File: system/core/mybigquery/bigquery_controller.go [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+	"time"
+
 	"app.modules/core/repository"
 	"app.modules/core/utils"
 	"cloud.google.com/go/bigquery"
-	"context"
-	"fmt"
 	"github.com/pkg/errors"
 	"google.golang.org/api/iterator"
 	"google.golang.org/api/option"
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/mybigquery/bigquery_controller.go:14]: PR範囲外なので見送り

File: system/core/workspaceapp/command_user.go [hunk 4]
```diff
@@ -118,31 +115,32 @@ func (app *WorkspaceApp) My(ctx context.Context, myOptions []utils.MyOption) err
 
 		// これ以降は書き込みのみ
 
-		replyMessage = i18n.T("common:sir", app.ProcessedUserDisplayName)
+		replyMessage = i18nmsg.CommonSir(app.ProcessedUserDisplayName)
 		currentRankVisible := userDoc.RankVisible
 		for _, myOption := range myOptions {
-			if myOption.Type == utils.RankVisible {
+			switch myOption.Type {
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/workspaceapp/command_user.go:121]: PR範囲外なので見送り

File: system/core/utils/parser.go [hunk 5]
```diff
@@ -195,10 +196,10 @@ func ReplaceEmojiCommandToText(fullString string) (string, string) {
 // FormatStringToParse はコマンド解析のために文字列を整形する
 func FormatStringToParse(fullString string) string {
 	// 全角スペースを半角に変換
-	fullString = strings.Replace(fullString, FullWidthSpace, HalfWidthSpace, -1)
+	fullString = strings.ReplaceAll(fullString, FullWidthSpace, HalfWidthSpace)
```
[the remark this answers is not in the document]
Comment (kani3camp) [reply system/core/utils/parser.go:199]: PR範囲外なので見送り
